spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepository.java

```diff
@@ -484,6 +486,14 @@ private boolean containsBranch(Git git, String label, ListMode listMode)
 		return false;
 	}
 
+	protected void warn(String message, Exception ex) {
+		if(logger.isDebugEnabled()) {
+			logger.warn(message, ex);
```

Comment:
It's a standard we follow, we don't check if one level is enabled, then log another. Feel free to log the warn AND then the debug.